thriftpy2/protocol/json.py

```diff
@@ -5,53 +5,57 @@
 import json
 import struct
 
+from thriftpy2._compat import u
 from thriftpy2.thrift import TType
 
 from .exc import TProtocolException
 
-INTEGER = (TType.BYTE, TType.I16, TType.I32, TType.I64)
-FLOAT = (TType.DOUBLE,)
-
 VERSION = 1
 
 
 def json_value(ttype, val, spec=None):
-    if ttype in INTEGER or ttype in FLOAT or ttype == TType.STRING:
-        return val
-
-    if ttype == TType.BOOL:
-        return True if val else False
-
-    if ttype == TType.STRUCT:
-        return struct_to_json(val)
-
-    if ttype in (TType.SET, TType.LIST):
-        return list_to_json(val, spec)
-
-    if ttype == TType.MAP:
-        return map_to_json(val, spec)
+    TTYPE_TO_JSONFUNC_MAP = {
```

Comment:
@ethe Do you think with there is negative performance impact if we move to map from branching?

Reply:
I think there is not much impact here.